UPchieve/web
# Reset wb

Links
-----
- Server repo PR: https://github.com/UPchieve/server/pull/483

Description
-----------
- Network call to reset the whiteboard
- Moved Zwibbler instantiation to a function `loadZwibbler` and call it upon resetting the whiteboard / receiving a socket event that the other user has reset the whiteboard
- Added an error message when deleting the whiteboard docstring from Redis fails
- Added a modal popup before confirming to reset the whiteboard

How to test
-----------

1. Log in to your volunteer account and log into your student account in another browser or in incognito mode.
2. Start a session as a student
   - Make sure your volunteer account has the subject included in their `subjects` array in the MongoDB document. These should be in the same format as the subject keys in `certifications`
   - Alternatively, copy and paste the session URL into the browser with the volunteer
3. Draw on the whiteboard as both student and volunteer
4. Press reset as either student or volunteer
5. Should be able to draw again on a new whiteboard screen

### Crash whiteboard and reset
This step only works if you're connected to our AWS bucket for uploading photos to the whiteboard
1. Once in a session with both student and volunteer, upload a photo as a student
2. Delete the photo as a volunteer
3. Press undo as a student and attempt to draw
  - This will crash the whiteboard. The student side will be unable to draw on the whiteboard and the volunteer side will only be able to use the whiteboard in read-only mode
4. Press reset as either student or volunteer
5. Should be able to draw again on a new whiteboard screen



Developer self-review checklist
-------------------------------
